File: Iniciante/PHP/exercise_1047.php
<?php

$tomorrowUnix = strtotime("+1 day");

//Format the timestamp into a date string
$tomorrow = date("Y-m-d", $tomorrowUnix);

$today = date("Y-m-d");

$valores = readline();

list ($horaInicial, $minutoInicial, $horaFinal, $minutoFinal) = sscanf($valores, "%d %d %d %d");

if ($horaInicial == $minutoInicial && $minutoInicial == $horaFinal && $horaFinal == $minutoFinal) {
    print_r("O JOGO DUROU 24 HORA(S) E 0 MINUTO(S)"."\n");
} else if ($horaInicial < $horaFinal || ($horaInicial == $horaFinal && $minutoInicial < $minutoFinal)) {
    $timeInicial = date_create(($horaInicial.":".$minutoInicial));
    $timeFinal = date_create(($horaFinal.":".$minutoFinal));

    $timeDiff = date_diff($timeInicial, $timeFinal);

    $horasDiff = $timeDiff->format('%H');
    $minutosDiff = $timeDiff->format('%i');

    print_r("O JOGO DUROU ".intval($horasDiff)." HORA(S) E ".$minutosDiff." MINUTO(S)"."\n");
} else if ($horaInicial > $horaFinal || ($horaInicial === $horaFinal && $minutoInicial > $minutoFinal)) {
    $timeInicial = date_create($today." ".(($horaInicial).":".$minutoInicial));
    $timeFinal = date_create($tomorrow." ".($horaFinal.":".$minutoFinal));

    $timeDiff = date_diff($timeInicial, $timeFinal);

    $horasDiff = $timeDiff->format('%H');
    $minutosDiff = $timeDiff->format('%i');

    print_r("O JOGO DUROU ".intval($horasDiff)." HORA(S) E ".$minutosDiff." MINUTO(S)"."\n");
}

//Concluído

File: Iniciante/Python/exercise_1047.py
from datetime import datetime
import datetime as original_datetime

horaInicial, minutoInicial, horaFinal, minutoFinal = [int(x) for x in input().split()]

now = datetime.now().strftime('%Y-%m-%d')

tempoInicial = [now, ' ', str(horaInicial), ':', str(minutoInicial)]
tempoFinal = [str(horaFinal), ':', str(minutoFinal)]

if horaInicial > horaFinal or (horaInicial == horaFinal and minutoInicial > minutoFinal):
    tomorrow = (datetime.now() + original_datetime.timedelta(days=1)).strftime('%Y-%m-%d')
    tempoFinal.insert(0, ' ')
    tempoFinal.insert(0, tomorrow)
else:
    tempoFinal.insert(0, ' ')
    tempoFinal.insert(0, now)


timeInicial = datetime.strptime(''.join(tempoInicial), '%Y-%m-%d %H:%M')
timeFinal = datetime.strptime(''.join(tempoFinal), '%Y-%m-%d %H:%M')
timeTotal = timeFinal - timeInicial
horaTotal = int(timeTotal.total_seconds()/60/60)
minutoTotal = int(timeTotal.total_seconds()%3600/60)

if horaInicial == minutoInicial and horaFinal == minutoFinal and horaInicial == horaFinal:
    print('O JOGO DUROU 24 HORA(S) E 0 MINUTO(S)')
else:
    print(''.join(['O JOGO DUROU ', str(horaTotal), ' HORA(S) E ', str(minutoTotal),' MINUTO(S)']))